dask/compose.py

```diff
@@ -0,0 +1,66 @@
+from uuid import uuid4
+import operator
+from functools import partial, wraps
+
+from toolz import merge
+
+from .optimize import cull
+from .context import _globals
+from . import threaded
+
+
+def get_uuid_dasks(v):
+    if isinstance(v, Value):
+        return v._uuid, v._dasks
+    else:
+        return v, []
+
+
+def applyfunc(func, *args, **kwargs):
+    func = partial(func, **kwargs)
+    uuids, dsks = zip(*map(get_uuid_dasks, args))
+    uuid = uuid4().hex
```

Comment:
Since these are imperatively built graphs, grabbing the file and line number might be more useful. I'm hesitant to use function name, as it can be gotten from the tasks already. No sense in having the same information twice.